src/test/java/com/diffmin/AppTest.java

```diff
@@ -197,25 +206,65 @@ void should_apply_pure_move_patches(TestResources sources) throws Exception {
+    }
+
     private static void runTests(TestResources sources) throws Exception {
         File f1 = sources.prevPath.toFile();
         File f2 = sources.newPath.toFile();
-        CtModel patchedCtModel = Main.patchAndGenerateModel(f1, f2);
-        CtModel expectedModel = SpoonUtil.buildModel(sources.newPath.toFile());
-        Optional<CtType<?>> firstType = expectedModel.getAllTypes().stream().findFirst();
-        if (firstType.isEmpty()) {
-            assertTrue(
-                    patchedCtModel.getAllTypes().stream().findFirst().isEmpty(),
-                    "Patched prev file is not empty");
-        } else {
-            CtType<?> retrievedFirstType = firstType.get();
-            CtCompilationUnit cu =
-                    retrievedFirstType
-                            .getFactory()
-                            .CompilationUnit()
-                            .getOrCreate(retrievedFirstType);
-            String patchedProgram = SpoonUtil.displayModifiedModel(patchedCtModel);
-            assertEquals(cu.prettyprint(), patchedProgram, "Prev file was not patched correctly");
+
+        Pair<Diff, CtModel> diffAndModel = Main.patchAndGenerateModel(f1, f2);
+        Diff diff = diffAndModel.getFirst();
+        CtModel patchedCtModel = diffAndModel.getSecond();
+        SpoonMapping mapping = SpoonMapping.fromGumTreeMapping(diff.getMappingsComp());
+
+        CtPackage prevPackage = SpoonUtil.buildModel(f1).getRootPackage();
+        CtPackage newPackage = SpoonUtil.buildModel(f2).getRootPackage();
+
+        List<String> insertedLines = Files.readAllLines(sources.insertedPaths);
+
+        Iterator<CtElement> it = patchedCtModel.getRootPackage().descendantIterator();
+
+        while (it.hasNext()) {
+            CtElement element = it.next();
+            if (isToIgnore(element, insertedLines)) {
+                continue;
+            }
+            CtPath path = element.getPath();
+            // for inserted elements
+
+            // Check if any of the path in metadata file match in patched program
+            if (insertedLines.stream()
+                    .anyMatch(line -> line.equals(path.toString()))) {
+                // Cases when the paths are same but the elements are different.
+                if (!path.evaluateOn(prevPackage).isEmpty()) {
+                    assertNotEquals(path.evaluateOn(prevPackage), path.evaluateOn(newPackage));
+                }
+                // The matched line should be there in new model and should match only once
+                assertEquals(1, path.evaluateOn(newPackage).size());
+            }
+            // There is no entry of the path of the element in the metadata file
+            else {
+                // If the element is not in prev model too, that means the path of the
+                // element has changed. We need to use `mapping`.
+                if (path.evaluateOn(prevPackage).size() == 0) {
```

Comment:
This test case was majorly added to handle matching of [this line](https://github.com/SpoonLabs/diffmin/blob/main/src/test/resources/insert/multiple-insert/NEW_TreeTraversal.java#L69). Although, it was there in the prev model, insertion of the [if block](https://github.com/SpoonLabs/diffmin/blob/main/src/test/resources/insert/multiple-insert/NEW_TreeTraversal.java#L66) shifts the index of the aforementioned statement by 1.